readme.md

```diff
@@ -86,17 +86,495 @@ Click the type names for complete docs.
+            this.config = { name, level };
+            Object.freeze(this.config);
+        }
+    }
+
+    const config: LoggerConfig = { name: 'MySuperApp', level: LogLevel.Debug };
+
+    const logger = new Logger(config);
+
+    // TypeScript Error: cannot assign to readonly.
+    logger.config.level = LogLevel.Error;
+
+    // We are able to edit config variable as we please.
+    config.level = LogLevel.Error;
+    ```
+    </details>
+
 - [`Pick<T, K>`](https://github.com/Microsoft/TypeScript/blob/2961bc3fc0ea1117d4e53bc8e97fa76119bc33e3/src/lib/es5.d.ts#L1422-L1427) - From `T`, pick a set of properties whose keys are in the union `K`.
+    <details>
+    <summary>
+        Example
+        <a href="https://typescript-play.js.org/?target=6#code/JYOwLgpgTgZghgYwgAgIJTMBAbFBvAKGWUzFwC5kBnMKUAcwG4iSALAVwFsAjEOYbJRp0QTFggD24COCG0GzAL4ECYAJ4AHFOkw4IABSgQAbsAgB3ZAF5k+rAGsAPDqy4ANMgBEpXJ+QAfLzAOHj4BTwA+ZgIAehjkACUZABNoZDhkbGAaZAkYdIxXCCpkdioGXJBsNRJgMhQ4EGTkVKoEOg1MKQA6Ahh2EAQukGQjJugXPUMTM3MqAAoNI1MLKkpJ3GmV8wBtAF0ASkoACQAVAFkAGQBRXE4ZMGRCYkkQHLhCvRKbZIkELge3XaEDgkFuEHu4HmnmSwGMngOzGIMAkUGQ81eOSWMwsuXy2O2VAOTxYxDi6Q0WiayAJsxIEjYDU+uCoLGUxCMYHYUBGH10LKUKkxjz5RW+oxSE2ZBmWswWOxYeFq9UonlOmggAGV2sBOiR2GBUcA4NgAISeDzBLi8fiCLwxOBUKgQMBUGKu7oAKw09D8ykOzCAA">
+            Playground
+        </a>
+    </summary>
+
+    ```typescript
+    interface Article {
+        title: string;
+        thumbnail: string;
+        content: string;
+    }
+
+    type ArticlePreview = Pick<Article, 'title' | 'thumbnail'>;
+
+    // Render a list of articles using only title and description.
+    function renderArticlePreviews(previews: ArticlePreview[]): HTMLElement {
+        const articles = document.createElement('div');
+        for (const preview of previews) {
+            // append preview to the articles
+        }
+        return articles;
+    }
+
+    const articles = renderArticlePreviews([
+        { title: 'TypeScript tutorial!', thumbnail: '/assets/ts.jpg', },
+    ]);
+    ```
+    </details>
+
 - [`Record<K, T>`](https://github.com/Microsoft/TypeScript/blob/2961bc3fc0ea1117d4e53bc8e97fa76119bc33e3/src/lib/es5.d.ts#L1429-L1434) - Construct a type with a set of properties `K` of type `T`.
+    <details>
+    <summary>
+        Example
+        <a href="https://typescript-play.js.org/?target=6#code/C4TwDgpgBAshC2AjCAnACgewM4Etg4wDssoBeKAIh0OFUIqgB9KATCANwgBsNIUHmFWgGMAFgFouEAIYsKAbgBQi6rRQAzacOhpUWIlADeiqFHU4UWYADlp8CAC4oVlNQDmS012lXb9py7unlAgMpYA8uoAogAefDgQhNpOhACuSKhKAL7KwkRWULR2TgBKEHkoLAA8cBno2HgExAA0ULqWRADaALoAfGRGJlCqdE49zUNsnDx8Y90Tpp1C5RJSshTdc4pZSooA9HtQ4akohTLwUKI+lxYQLFDSUIQQAO5Q9nUAdIpF8J8jKEInzAqSwogAFIYzBZfHZHJQAFIYUT0VreWH+SgAEQwEAorVC0gi0TiqASSXhAAYoFkAJS7A5QMoVe7qDAobQkEAYVKFDDDQiNaRcHAAL2gwq4UAw6kKomgYBQvFQoCgAGsICAsPtDgAVcAQADKwlcYGAUCiKCVKCcyzEkhkcigiuVKFVOBI8A9uEIbkUeWI5t+UXgZpApXK7OqtWQ9Vw+HyrTSXC4-XIxlMpgBhBSqRTC0zU24rtz+e28iAA">
+            Playground
+        </a>
+    </summary>
+
```

Comment:
For each type it would be useful to include a non-technical sentence of what it does here. ``` Construct a type with a set of properties `K` of type `T` ``` describes the type in type-system terms, but not in human terms.